---
title: "R Notebook"
output: html_notebook
---

#Import tables
```{r}
setwd("/Users/karlavasco/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip")

library(readxl)
#read in otu table (smartchip normalized abundance)
otu_table <- read_excel("smartchip_phyloseq_tables.xlsx", sheet = "abundance")

library(dplyr)
#converting the abundance table to matrix (required to merge with taxonomy table)
otu_table <- otu_table %>% remove_rownames %>% column_to_rownames(var="GENE_ID") 
otu_table=as.matrix(otu_table,rownames=TRUE)

#multiplying abundance by 10^9 so we can calculate diversity measures (it won't be possible with decimals)
otu_table <- otu_table*10^9
mode(otu_table) <- "integer"

#converting empty values to 0 (in case they exist)
otu_table[is.na(otu_table)] <- 0

#read in taxonomy (gene ID for smartchip)
taxonomy <- read_excel("smartchip_phyloseq_tables.xlsx", sheet = "gene_ID")

#setting taxonomy row_names for GENE_ID (that will enable merging with the abundance table)
taxonomy <- taxonomy %>% remove_rownames %>% column_to_rownames(var="GENE_ID")
taxonomy=as.matrix(taxonomy)

#read in metadata
metadata <- read_excel("smartchip_phyloseq_tables.xlsx", sheet = "metadata")

#setting metadata row_names for sample ID (that will enable merging with the other tables)
metadata <- metadata %>% remove_rownames %>% column_to_rownames(var="Sample_ID")

#Changing the levels of source and group to make them better organized in the plots
metadata <- metadata %>% mutate(Source = fct_relevel(Source,"Meconium","Infant 7 days", "Mother"))
metadata <- metadata %>% mutate(Group = fct_relevel(Group,"Control","Sepsis"))

library("phyloseq")

#import all tables as phyloseq objects
OTU = otu_table(otu_table,taxa_are_rows=TRUE)
TAX = tax_table(taxonomy)
META = sample_data(metadata)

# Sanity checks for consistent OTU names
taxa_names(TAX)
taxa_names(OTU)

# Same sample names
sample_names(OTU)
sample_names(META)

#Making phyloseq object
physeq=phyloseq(OTU,TAX,META)
```

#Filtering samples by Source for comparisons within source
```{r}
physeq_meconium <-  subset_samples(physeq, Source %in% c("Meconium"))
physeq_7days <-  subset_samples(physeq, Source %in% c("Infant 7 days"))
physeq_mother <-  subset_samples(physeq, Source %in% c("Mother"))
```


#ANCOM-BC
```{r}
library(microbiome)
library(nloptr)
library(ANCOMBC)
```

#ANTIBIOTIC USE 
Any or non
##Gene
```{r}
#ANCOMBC analysis comparison between treatment groups
out = ancombc(phyloseq = physeq, formula = "Any_Ab_use",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Any_Ab_use", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_gene_Any_Ab_use.csv",sep = ",", row.names = F) 
```

##Class
```{r}
# Aggregate the abundance to the "Class" level
physeq = tax_glom(physeq, "Class")

#ANCOMBC analysis comparison between groups
out = ancombc(phyloseq = physeq, formula = "Any_Ab_use",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Any_Ab_use", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_class_Any_Ab_use.csv",sep = ",", row.names = F) 
```

#Antibiotic use by Any_Ab_use
#MECONIUM
Between Any_Ab_use
##Gene
```{r}
# Aggregate the abundance to the "Gene" level
physeq=physeq_meconium

#ANCOMBC analysis comparison between Any_Ab_uses
out = ancombc(phyloseq = physeq, formula = "Any_Ab_use",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Any_Ab_use", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_gene_meconium_Any_Ab_use.csv",sep = ",", row.names = F) 
```

##Class
```{r}
# Aggregate the abundance to the "Class" level
physeq = tax_glom(physeq, "Class")

#ANCOMBC analysis comparison between Any_Ab_uses
out = ancombc(phyloseq = physeq, formula = "Any_Ab_use",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Any_Ab_use", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_class_meconium_Any_Ab_use.csv",sep = ",", row.names = F) 
```


#INFANT 7 DAYS
Between Any_Ab_uses
##Gene
```{r}
# Aggregate the abundance to the "Gene" level
physeq=physeq_7days

#ANCOMBC analysis comparison between Any_Ab_uses
out = ancombc(phyloseq = physeq, formula = "Any_Ab_use",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Any_Ab_use", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_gene_infant7days_Any_Ab_use.csv",sep = ",", row.names = F) 
```


##Class
```{r}
# Aggregate the abundance to the "Class" level
physeq = tax_glom(physeq, "Class")

#ANCOMBC analysis comparison between Any_Ab_uses
out = ancombc(phyloseq = physeq, formula = "Any_Ab_use",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Any_Ab_use", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_class_infant7days_Any_Ab_use.csv",sep = ",", row.names = F) 
```

#MOTHER
Between Any_Ab_uses
##Gene
```{r}
# Aggregate the abundance to the "Gene" level
physeq=physeq_mother

#ANCOMBC analysis comparison between Any_Ab_uses
out = ancombc(phyloseq = physeq, formula = "Any_Ab_use",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Any_Ab_use", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_gene_mother_Any_Ab_use.csv",sep = ",", row.names = F) 
```

##Class
```{r}
# Aggregate the abundance to the "Class" level
physeq = tax_glom(physeq, "Class")

#ANCOMBC analysis comparison between Any_Ab_uses
out = ancombc(phyloseq = physeq, formula = "Any_Ab_use",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Any_Ab_use", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_class_mother_Any_Ab_use.csv",sep = ",", row.names = F) 
```

############################################################
#GROUPS 
Sepsis vs. control
##Gene
```{r}
#ANCOMBC analysis comparison between treatment groups
out = ancombc(phyloseq = physeq, formula = "Group",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Group", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_gene_groups.csv",sep = ",", row.names = F) 
```

##Class
```{r}
# Aggregate the abundance to the "Class" level
physeq = tax_glom(physeq, "Class")

#ANCOMBC analysis comparison between groups
out = ancombc(phyloseq = physeq, formula = "Group",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Group", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_class_groups.csv",sep = ",", row.names = F) 
```


#MECONIUM
Between groups
##Gene
```{r}
# Aggregate the abundance to the "Gene" level
physeq=physeq_meconium

#ANCOMBC analysis comparison between groups
out = ancombc(phyloseq = physeq, formula = "Group",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Group", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_gene_meconium_groups.csv",sep = ",", row.names = F) 
```

##Class
```{r}
# Aggregate the abundance to the "Class" level
physeq = tax_glom(physeq, "Class")

#ANCOMBC analysis comparison between groups
out = ancombc(phyloseq = physeq, formula = "Group",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Group", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_class_meconium_groups.csv",sep = ",", row.names = F) 
```


#INFANT 7 DAYS
Between groups
##Gene
```{r}
# Aggregate the abundance to the "Gene" level
physeq=physeq_7days

#ANCOMBC analysis comparison between groups
out = ancombc(phyloseq = physeq, formula = "Group",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Group", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_gene_infant7days_groups.csv",sep = ",", row.names = F) 
```

##Class
```{r}
# Aggregate the abundance to the "Class" level
physeq = tax_glom(physeq, "Class")

#ANCOMBC analysis comparison between groups
out = ancombc(phyloseq = physeq, formula = "Group",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Group", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_class_infant7days_groups.csv",sep = ",", row.names = F) 
```
#MOTHER
Between groups
##Gene
```{r}
# Aggregate the abundance to the "Gene" level
physeq=physeq_mother

#ANCOMBC analysis comparison between groups
out = ancombc(phyloseq = physeq, formula = "Group",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Group", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_gene_mother_groups.csv",sep = ",", row.names = F) 
```

##Class
```{r}
# Aggregate the abundance to the "Class" level
physeq = tax_glom(physeq, "Class")

#ANCOMBC analysis comparison between groups
out = ancombc(phyloseq = physeq, formula = "Group",
              p_adj_method = "holm", zero_cut = 0.90, lib_cut = 0,
              group = "Group", struc_zero = F, neg_lb = FALSE,
              tol = 1e-05, max_iter = 100, conserve = TRUE,
              alpha = 0.05, global = TRUE)

#ANCOMBC results as a list
res = out$res

#ANCOMBC results as a table
ancom_results = res %>% as_tibble()

#Filtering only significant results
ancom_signif_list <- ancom_results %>% filter(p_val <= 0.05)

#Adding feature names to the table
ancom_signif_names <- data.frame(sapply(ancom_signif_list,c))
row.names(ancom_signif_names) <- rownames(ancom_signif_list$beta) 
ancom_signif_names <- merge(physeq@tax_table,ancom_signif_names, by = 0)

#Exporting significant results as a csv file
write.table(ancom_signif_names, file="~/OneDrive\ -\ Michigan\ State\ University/Zhang_lab/Mila_project/tables/smartchip/ancombc/ANCOM_class_mother_groups.csv",sep = ",", row.names = F) 
```

